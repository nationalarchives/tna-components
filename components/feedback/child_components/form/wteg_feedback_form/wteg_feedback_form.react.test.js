import React from 'react';
import WTEGFeedbackForm from './wteg_feedback_form.react';
import TestRenderer, { create } from 'react-test-renderer';

// Snapshot 1
it('renders correctly', () => {
	const tree = TestRenderer.create(<WTEGFeedbackForm />).toJSON();
	expect(tree).toMatchSnapshot();
});

// Unit test
describe('Test state', () => {
	const component = create(<WTEGFeedbackForm />);
	const instance = component.getInstance();
	const {
		initialQuestion,
		displayCommentBox,
		guideHelpful,
		message,
		displayForm
	} = instance.state;
	it('checking the initial state', () => {
		expect(initialQuestion).toBeTruthy();
		expect(displayCommentBox).toBeFalsy();
		expect(guideHelpful).toBeFalsy();
		expect(message).toBeFalsy();
		expect(displayForm).toBeTruthy();
	});
});
