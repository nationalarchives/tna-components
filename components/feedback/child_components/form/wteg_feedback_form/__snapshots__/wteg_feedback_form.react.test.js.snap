// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <fieldset
      className="initial-question"
      id={null}
    >
      <legend>
        Did you find this helpful?
      </legend>
      <button
        onClick={[Function]}
        type="button"
      >
        Yes
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        No
      </button>
    </fieldset>
  </form>
</div>
`;
